apps/messages/sw.js

```diff
@@ -75,26 +75,33 @@ self.onsystemmessage = async (event) => {
     return;
   }
 
-  // let handler = event.data.webActivityRequestHandler();
-  // let source = handler.source;
+  if (event.name === "activity") {
+    let handler = event.data.webActivityRequestHandler();
+    let source = handler.source;
 
-  // let resolver;
-  // let promise = new Promise((resolve) => {
-  //   resolver = resolve;
-  // });
-  // event.waitUntil(promise);
+    if (source.name !== "new") {
+      console.error(`Unexpected activity: ${source.name}`);
+      return;
+    }
+
+    let resolver;
+    let promise = new Promise((resolve) => {
+      resolver = resolve;
+    });
+    event.waitUntil(promise);
 
-  // // console.log(`ZZZ Contact activity data: ${JSON.stringify(source.data)}`);
+    console.log(`MMM Contact activity data: ${JSON.stringify(source.data)}`);
 
-  // const allClients = await clients.matchAll({
-  //   includeUncontrolled: true,
-  // });
-  // if (allClients.length > 0) {
-  //   let win = allClients[0];
-  //   win.postMessage(source.data);
-  // } else {
-  //   let win = await clients.openWindow("/index.html");
-  //   win.postMessage(source.data);
-  // }
-  // resolver();
+    const allClients = await clients.matchAll({
+      includeUncontrolled: true,
+    });
+    if (allClients.length > 0) {
+      let win = allClients[0];
```

Comment:
Please add a call to `win.focus();` here to make sure that if the app is currently running but not in the foreground it will be brought as the foreground one.